### PR TITLE
loadable_apps/binary_update: Run update aging test with random binary

### PR DESCRIPTION
**Pull request by jeongarmy**
Run update aging test with Micom or Wifi binary randomly.

### CODE REVIEW

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 1]
```diff
@@ -644,26 +640,16 @@ void binary_update_test(void)
...
-		}
-
-		sleep(5);
-
-		ret = binary_update_reload(APP_NAME);
-		if (ret != OK) {
-			return;
+		randval = rand();
+		if (randval % 2 == 0) {
+			/* Iterate 'Micom' binary update */
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:648]: This sample iterates the binary update by randomly, but in the comments it looks that it iterates micom binary update only.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 2]
```diff
@@ -644,26 +640,16 @@ void binary_update_test(void)
...
+			binary_update_new_version_test(APP_NAME);
+		} else {
+			/* Iterate 'Wifi' binary update by itself */
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:651]: Same as above.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 3]
```diff
@@ -484,24 +485,19 @@ static void binary_update_new_version_test(void)
 		return;
 	}
 
-	ret = binary_update_reload(APP_NAME);
+	ret = binary_update_reload(bin_name);
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:488]: As discussed before, please change the function name. This is not reload.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 4]
```diff
@@ -484,24 +485,19 @@ static void binary_update_new_version_test(void)
...
 	if (ret != OK) {
 		return;
 	}
 
-	sleep(2);
+	sleep(3);
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:493]: It means that after reload, should we wait some seconds?
If then, we should guide how to use.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 5]
```diff
@@ -484,24 +485,19 @@ static void binary_update_new_version_test(void)
...
 		return;
 	}
 
-	ret = binary_update_check_test_result(&pre_bin_info, &cur_bin_info, DOWNLOAD_VALID_BIN);
-	if (ret == OK) {
-		/* Unlink binary file with old version */
-		snprintf(filepath, BINARY_PATH_LEN, "%s/%s_%u", BINARY_DIR_PATH, APP_NAME, (uint32_t)pre_bin_info.version);
-		unlink(filepath);
-	}
+	binary_update_check_test_result(&pre_bin_info, &cur_bin_info, DOWNLOAD_VALID_BIN);
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:500]: In previous, old version binary was unlinked. But in here, you removed the unlink.
Who will unlink the old one? Is it not needed?

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 6]
```diff
@@ -35,6 +35,7 @@
 
 /* App binary information for update test */
 #define APP_NAME                   "micom"
+#define MYAPP_NAME                 "wifi"
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:38]: I think `MICOM_APP_NAME` and `WIFI_APP_NAME` are better.

Comment (davidfather) [reply loadable_apps/loadable_sample/wifiapp/binary_update.c:38]: How about `app1` and `app2`? `micom` and `wifi` are too specific.

**Review by sunghan-chang — DISMISSED** [review]

**Comment by sunghan-chang** [comment]
Could you resolve the conflict?

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 7]
```diff
@@ -583,7 +579,7 @@ static void binary_update_run_tests(void)
 	binary_update_register_state_changed_callback();
 
 	/* 4. Reload test with new version. */
-	binary_update_new_version_test();
+	binary_update_new_version_test(APP1_NAME);
 
 	/* 5. Unregister registered callback. */
 	binary_update_unregister_state_changed_callback();
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:585]: If registering callback fails, then we don't need to unregister the callback.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 8]
```diff
@@ -452,56 +453,51 @@ static void binary_update_same_version_test(void)
...
 		printf("Get binary info FAIL, ret %d\n", ret);
 		close(ret);
 	}
 }
 
-static void binary_update_new_version_test(void)
+static int binary_update_new_version_test(char *bin_name)
 {
 	int ret;
 	char filepath[BINARY_PATH_LEN];
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:473]: `filepath` is not needed anymore. Please remove this.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 9]
```diff
@@ -645,25 +641,19 @@ void binary_update_test(void)
...
-			return;
-		}
-
-		/* Copy current binary with a higher version and valid crc. */
-		ret = binary_update_download_binary(&bin_info, DOWNLOAD_VALID_BIN);
-		if (ret != OK) {
-			return;
+		/* Iterate update test with random binary */
+		randval = rand();
+		if (randval % 2 == 0) {
```
Comment (Taejun-Kwon) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:649]: To make a more easier than now, actually powercut stuff  works random time both of turn-on & turn-off, so I think random is unnecessary here, I mean Test Equipment already uses its own random logic.
Testing can be done sequentially.

Comment (jeongarmy) [reply loadable_apps/loadable_sample/wifiapp/binary_update.c:649]: This is for random scenario, not random timing for update. It makes it to run micom update or wifi update ramdomly,.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**